GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -79,6 +88,21 @@ private void OkClick(object sender, EventArgs e)
                     return;
                 }
 
+                HashSet<string> selectedWorktreeBranches = selectedBranches
+                    .Where(branch => _worktreeBranches.Contains(branch.Name))
+                    .Select(branch => branch.LocalName).ToHashSet();
+                if (selectedWorktreeBranches.Count > 0)
+                {
+                    string output = Module.GitExecutable.GetOutput(GitCommandHelpers.ListWorkTreeCmd());
+                    List<GitWorkTree> worktrees = ListWorkTreeOutputParser.Parse(output)
+                        .Where(w => selectedWorktreeBranches.Contains(w.BranchName)).ToList();
+                    string paths = string.Join(Environment.NewLine, worktrees
+                        .Select(worktree => string.Format(_branchPathMessage.Text, worktree.BranchName, worktree.Path)));
```

Comment:
We are materialising `worktrees` collection then to iterate it again with `Select`.
This can be reworked to reduce allocations